### PR TITLE
(NETDEV-30) Enhance syslog, RADIUS, TACACS types

### PR DESCRIPTION
**Pull request by shermdog**
This commit adds the following:
syslog_server
- port

syslog_settings
- console
- monitor
- source_interface
- vrf

tacacs_global
- source_interface
- vrf

radius_global
- source_interface
- vrf

### CODE REVIEW

File: lib/puppet/type/syslog_settings.rb [hunk 1]
```diff
@@ -20,8 +20,48 @@
...
+
+    validate do |value|
+      if (value.to_s.match('^[0-7]$') || value == 'unset') then super(value)
+      else fail "value #{value.inspect} is invalid, must be 0-7 or 'unset'"
+      end
+    end
+  end
+
+  newproperty(:source_interface) do
+    desc 'Source interface to send syslog data from, e.g. "ethernet 2/1"'
```
Comment (mikewiebe) [inline lib/puppet/type/syslog_settings.rb]: Does this property support `unset`?

**Comment by johnduarte** [comment]
@shermdog what is the blocker for this PR?

**Comment by shermdog** [comment]
@johnduarte we have to coordinate the changes with multiple vendors.  That process is ongoing.

**Comment by mikewiebe** [comment]
👍 Changes look good to me.

**Pull request merged**